### PR TITLE
FEAT: ClickHouse: support joining on different columns.

### PR DESCRIPTION
**Pull request by Lunaticus7**
It seems that for some historical reasons ClickHouse backend banned joining on different columns, and that has been supported by ClickHouse now. So i removed the vailidation and added some codes for that (refer to ibis SQL backend).

### CODE REVIEW

File: ibis/backends/clickhouse/tests/test_select.py [hunk 1]
```diff
@@ -265,43 +267,43 @@ def test_non_equijoin(alltypes):
...
-    ],
+    ('join_type_and_clause', 'join_keys'),
+    product(
+        [
+            ('any_inner_join', 'ANY INNER JOIN'),
+            ('inner_join', 'ALL INNER JOIN'),
+            ('any_left_join', 'ANY LEFT JOIN'),
+            ('left_join', 'ALL LEFT JOIN'),
+        ],
+        [('playerID',), ('playerID', 'awardID'),],  # noqa: E231
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/clickhouse/tests/test_select.py:279]: They serve different purposes. `black` is for formatting, `flake8` will lint a number of things. The flake8 configuration is probably not maximally black compatible. There are also scenarios where black and flake8 will conflict, even if configured for maximum compatibility. Chalk it up to the Python community having 10 different tools and conventions for formatting code.

File: ibis/backends/clickhouse/tests/test_select.py [hunk 2]
```diff
@@ -265,43 +267,43 @@ def test_non_equijoin(alltypes):
...
+        f'  {join_clause} {db.name}.`awards_players` t1\n'
+        f'{join_keys_str}'
     )
 
     assert ibis.clickhouse.compile(expr) == expected
-    assert len(con.execute(expr))
+    try:
+        con.execute(expr)
+    except clickhouse_driver.errors.ServerException:
+        pytest.fail('ClickHouse raise a `ServerException` error.')
```
Comment (cpcloud) [inline ibis/backends/clickhouse/tests/test_select.py:306]: Consider letting this fail, or if it must not fail then at least format the exception into the `fail` strings. It's nice to know that a `ServerException` was raised, but it's even better to know what that exception is.

File: ibis/backends/clickhouse/compiler.py [hunk 3]
```diff
@@ -84,14 +84,35 @@ def get_result(self):
...
+            fmt_preds = []
+            npreds = len(preds)
+            using_clause_flag = True
+            for pred in preds:
+                op = pred.op()
+                if (
+                    not isinstance(op, ops.Equals)
+                    or op.args[0].get_name() != op.args[1].get_name()
+                ):
+                    using_clause_flag = False
```
Comment (cpcloud) [inline ibis/backends/clickhouse/compiler.py:98]: Can we avoid using the `USING` keyword here? It's adding a ton of complexity to this code and if it isn't necessary then we shouldn't use it.

File: ibis/backends/clickhouse/registry.py [hunk 4]
```diff
@@ -471,11 +471,10 @@ def _table_column(translator, expr):
...
 
-    # TODO(kszucs): table aliasing is partially supported
-    # if ctx.need_aliases():
-    #     alias = ctx.get_ref(table)
-    #     if alias is not None:
-    #         quoted_name = '{0}.{1}'.format(alias, quoted_name)
+    if ctx.need_aliases():
+        alias = ctx.get_ref(table)
+        if alias is not None:
+            quoted_name = '{0}.{1}'.format(alias, quoted_name)
```
Comment (cpcloud) [inline ibis/backends/clickhouse/registry.py:477]: Use an f-string here since you're already touching this code

**Review by cpcloud — CHANGES_REQUESTED** [review]
Looks mostly good, except for the `USING` comment.

File: ibis/backends/clickhouse/compiler.py (hunk 3 above)
Comment (Lunaticus7) [reply ibis/backends/clickhouse/compiler.py]: When stop using the `USING` keyword, there is no need to override this method, so i remove it and some othere duplicated codes.

File: ibis/backends/clickhouse/tests/test_select.py (hunk 2 above)
Comment (Lunaticus7) [reply ibis/backends/clickhouse/tests/test_select.py]: I have remove the try caluse.

File: ibis/backends/clickhouse/tests/test_select.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lunaticus7) [reply ibis/backends/clickhouse/tests/test_select.py]: thanks for explaining.

[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/clickhouse/tests/test_select.py]: For sure!

File: ibis/backends/clickhouse/compiler.py (hunk 3 above)
Comment (cpcloud) [reply ibis/backends/clickhouse/compiler.py]: Awesome. Loving the code deletion.

**Review by cpcloud — APPROVED** [review]
Net negative lines of code. Ship it.

**Pull request merged**